pyuvdata/beamfits.py

```diff
@@ -50,276 +50,276 @@ def read_beamfits(self, filename, run_check=True, check_extra=True,
+                    self.data_array = data
+
+                # Note: This axis is called STOKES by analogy with the equivalent uvfits axis
+                # However, this is confusing because it is NOT a true Stokes axis,
+                #   it is really the polarization axis.
+                if primary_header.pop('CTYPE' + str(ax_nums['feed_pol'])).lower().strip() == 'stokes':
+                    self.Npols = primary_header.pop('NAXIS' + str(ax_nums['feed_pol']))
+
+                self.polarization_array = np.int32(uvutils.fits_gethduaxis(primary_hdu,
+                                                                           ax_nums['feed_pol']))
+                self.set_power()
+            elif self.beam_type == 'efield':
+                self.set_efield()
+                if n_dimensions < n_efield_dims:
+                    raise ValueError('beam_type is efield and data dimensionality is too low')
+                complex_arrs = np.split(data, 2, axis=0)
+                self.data_array = np.squeeze(complex_arrs[0] + 1j * complex_arrs[1], axis=0)
+                if primary_header.pop('CTYPE' + str(ax_nums['feed_pol'])).lower().strip() == 'feedind':
+                    self.Nfeeds = primary_header.pop('NAXIS' + str(ax_nums['feed_pol']))
+                feedlist = primary_header.pop('FEEDLIST', None)
+                if feedlist is not None:
+                    self.feed_array = np.array(feedlist[1:-1].split(', '))
+            else:
+                raise ValueError('Unknown beam_type: {type}, beam_type should be '
+                                 '"efield" or "power".'.format(type=self.beam_type))
+
+            self.data_normalization = primary_header.pop('NORMSTD', None)
+
+            self.telescope_name = primary_header.pop('TELESCOP')
+            self.feed_name = primary_header.pop('FEED', None)
+            self.feed_version = primary_header.pop('FEEDVER', None)
+            self.model_name = primary_header.pop('MODEL', None)
+            self.model_version = primary_header.pop('MODELVER', None)
+
+            # shapes
+            if primary_header.pop('CTYPE' + str(ax_nums['freq'])).lower().strip() == 'freq':
+                self.Nfreqs = primary_header.pop('NAXIS' + str(ax_nums['freq']))
+
+            if n_dimensions > ax_nums['spw'] - 1:
+                if primary_header.pop('CTYPE' + str(ax_nums['spw'])).lower().strip() == 'if':
+                    self.Nspws = primary_header.pop('NAXIS' + str(ax_nums['spw']), None)
+                    # subtract 1 to be zero-indexed
+                    self.spw_array = uvutils.fits_gethduaxis(primary_hdu, ax_nums['spw']) - 1
+
+            if n_dimensions > ax_nums['basisvec'] - 1:
+                if primary_header.pop('CTYPE' + str(ax_nums['basisvec'])).lower().strip() == 'vecind':
+                    self.Naxes_vec = primary_header.pop('NAXIS' + str(ax_nums['basisvec']), None)
+
+            if (self.Nspws is None or self.Naxes_vec is None) and self.beam_type == 'power':
```

Comment:
Aha, heres the rest of the bits which were missing above.  This diff is hopeless. Lets just assume everythings fine.